### PR TITLE
ExportGen IR conversion

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: python/hail/methods/impex.py [hunk 1]
```diff
@@ -60,46 +72,93 @@ def export_gen(dataset, output, precision=4):
...
+    if id1 is None:
+        require_col_key_str(dataset, "export_gen")
+        id1 = dataset.col_key[0]
+
+    if id2 is None:
+        require_col_key_str(dataset, "export_gen")
+        id2 = dataset.col_key[0]
+
+    if missing is None:
+        missing = hl.float64(0)
```
[the remark this answers is not in the document]
Comment (jigold) [reply python/hail/methods/impex.py]: Didn't work. I reverted it back to `hl.float64()`.

[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/methods/impex.py]: where was the problem coming from?

**Comment by tpoterba** [comment]
oh, the analyze. fair enough.

**Comment by tpoterba** [comment]
ah, you're totally right. Didn't read the interface diff closely enough.

File: src/main/scala/is/hail/io/gen/ExportGen.scala [hunk 2]
```diff
@@ -1,91 +1,144 @@
...
+
+        if (spaceRegex.findFirstIn(id1).isDefined)
+          fatal(s"Invalid 'id1' found -- no white space allowed: '$id1'")
+        if (spaceRegex.findFirstIn(id2).isDefined)
+          fatal(s"Invalid 'id2' found -- no white space allowed: '$id2'")
+        if (missing < 0 || missing > 1)
+          fatal(s"'missing' values must be in the range [0, 1]. Found $missing for ($id1, $id2).")
+
+        s"$id1 $id2 $missing"
+      }.toList)
```
Comment (tpoterba) [inline src/main/scala/is/hail/io/gen/ExportGen.scala]: > .toList

😱

**Comment by tpoterba** [comment]
ah, you're totally right. Didn't read the interface diff closely enough.

**Pull request merged**